### DISCUSSION TITLE
Unsupported media type

### DISCUSSION THREAD

**Discussion by jeremyisatrecharm** [deepgram/community#327] [General help] [answered]
### Which Deepgram product are you using?

None

### Details

I am using the Node.JS library and calling preRecorded via "nova".

We are submitting audio to get transcribed. Sometimes the audio files are empty -- we can't tell if there are folks talking until ya'll transcribe. This worked fine up until today, yesterday or the day before (I have had no related code changes) where I now get an error response:

```
DG: {"err_code":"Unsupported Media Type","err_msg":"remote server failed to offer audio data. Content-Type of remote server's response was: application/zip"
```

My hunch is this is happening when I am submit audio that is silent. Returning an error seems reasonable from you (honestly, as does charging me for transcribing empty audio - your choice!). But, I wanted to be certain this is what is happening and I couldn't find any documentation.

If this is the new behavior, how can I differentiate between this a "silent audio upload" error and another more genuine similar error returned by the API?

**Comment by jjmaldonis** [comment]
Hey @jeremyisatrecharm, we made a change to return a more useful error message if the content-type of the payload is not supported. It's possible this edge case with .zip files was impacted, although I was not aware that we accepted .zip files ever.

Can you please add copy-paste the code that was working a few days ago? I'll take a look.

**Comment by jeremyisatrecharm** [comment]
This is an odd thing... We are sending `mp3s`... 

```
    const result = await  new Deepgram(process.env.DEEPGRAM_API_KEY).transcription.preRecorded(
      { url: fileUrl },
      {
        smart_format: true,
        model: "nova",
      }
    );
    if (!result.results) {
      throw new Error("Deepgram response is missing results");
    }

    return result.results;
```

If it's helpful here is a `request_id` that failed for me with the above error: 
```
{
"request_id":"88253e8c-5d29-42ce-af5f-9d6bed035fb7"
}
```

**Reply by jjmaldonis** [reply]
Can you check the headers associated with your `fileUrl`? You can get the headers using `curl -i <fileUrl>`, where `fileUrl` is the one you linked on line 2 of your codebock.

The curl response should include all the headers sent by your server, and we are specifically looking for the `Content-Type` header. Does your server send `Content-Type: application/zip`, even though it's sending .mp3 files?

**Comment by jeremyisatrecharm** [comment]
oh wow, AWS _is_ returning `Content-Type: application/zip`. Note this is a `.mp3` file.... 

I can try to change the uploaded MIME type to mp3; though, will the API return `remote server failed to offer audio data.` for silent mp3s or a transcription of `""` (the behavior I saw before).

**Reply by jjmaldonis** [reply]
Good to know! I'm very curious why AWS is sending that content type. If you figure it out, I'd love to hear about any insights you have.

Silent .mp3s will return an empty transcript of `""` - just like you saw before.

**Reply by jeremyisatrecharm** [reply, marked as answer]
OK I am pretty sure _all_ of my transcribe calls were failing. I was putting a `.mp3` file on S3 and did not assign a MIME type. I guess AWS by default was giving the file a zip MIME type 🤷‍♂️ . I suppose Deepgram started erroring on non-audio MIME types.

This is all sorted - thanks.
